### DISCUSSION TITLE
Private repository contributions not showing on profile for non-master branches

### DISCUSSION THREAD

**Discussion by harish18092002** [community/community#138788] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

I'm encountering an issue where contributions to a new branch in my private repository are not appearing in my GitHub profile's contribution history. Here are the details:

Contributions to the master branch of this private repository do show up correctly on my profile.
I've enabled the 'Allow private contributions' option in my GitHub settings.
I've confirmed that the email address associated with my git CLI matches my GitHub account.
The issue only occurs with non-master branches in this private repository.

I've double-checked all the common troubleshooting steps, but the problem persists. Has anyone else experienced this issue or know of a solution? Any insights or suggestions would be greatly appreciated.
Additionally, I'm curious if this is expected behavior or if it might be a bug in how GitHub tracks contributions from private repositories across different branches.

### My Commit history 

<img width="1791" alt="Screenshot 2024-09-16 at 5 17 05 PM 1" src="https://github.com/user-attachments/assets/3f47d581-0c0d-4585-a36f-6a6194d72434">

### My contributions history 

I've made numerous commits over the past three days, but my contribution history only displays a single commit from the main branch. This single commit is from the same repository where I've been actively working on other branches. The contributions from these other branches are not being reflected in my history, despite being part of the same repository.

<img width="1012" alt="Screenshot 2024-09-16 at 5 21 39 PM" src="https://github.com/user-attachments/assets/f2533fa6-291d-4da5-9cc7-c035cc989052">

**Comment by davevad93** [comment, marked as answer]
Hi @harish18092002 ,

That's normal, only commits made in the default branch are counted as contributions (for both public and private repos), when you'll merge the changes to the main branch, all the commits you made plus the pull request will be counted as contributions and will be shown in your activity graph.

For more info, take a look at the docs:

-  [Why are my contributions not showing up on my profile?](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile)

**Reply by harish18092002** [reply]
Yeah @davevad93 ....It worked perfectly,I really appreciate your help with this issue. Thank you so much for taking the time to assist me!!

**Reply by davevad93** [reply]
You're welcome @harish18092002 , have a nice day! 👍

**Comment by ShadowCoder02** [comment]
Hey @harish18092002,

That’s completely normal! Only commits made in the default branch are recognized as contributions (this applies to both public and private repositories). Once you merge your changes into the main branch, all the commits you’ve made, along with the pull request, will be counted as contributions and reflected in your activity graph.
